### PR TITLE
Windows compatibility

### PR DESCRIPTION
**Pull request by mehrdadn**
~Still incomplete, but if the tests pass, it'd be nice if this could be merged, since there are breaking changes with the codebase and I keep having to update the code every time.~

~Note that this adds several submodules to the repository.~

Updates regarding Windows compatibility.

### CODE REVIEW

**Comment by robertnishihara** [comment]
We should probably close this PR since it is pretty out of date. I'd also like to delete the branch so that the only branches on the main repository are for the current master, the github pages webpage, and release tags.

We definitely don't want to lose the code, though I think github will preserve it. But in the worst case, I have the branch on my fork https://github.com/robertnishihara/ray/tree/win-compat.

Thoughts about this?

**Comment by mehrdadn** [comment]
I wouldn't rely on GitHub keeping anything deleted around (and I wouldn't rely on personal forks since they're too easy to delete), but I just sent you an email with the code, so yup feel free to close/delete.

**Comment by robertnishihara** [comment]
@mehrdadn did you intentionally reopen this?

**Comment by mehrdadn** [comment]
@robertnishihara Yeah haha

**Review by rkooo567 — APPROVED** [review]

**Comment by rkooo567** [comment]
@mehrdadn Can you ping me after you merge this PR?

File: doc/source/installation.rst [hunk 1]
```diff
@@ -155,6 +173,20 @@ the dashboard. For example,
...
+----------------------------
+
+Windows support is currently limited and "alpha" quality.
+Bugs, process/resource leaks, or other incompatibilities may exist under various scenarios.
+Unusual, anattended, or production usage is **not** recommended.
+
+If you encounter any issues, please check the following:
+
+- The Known Issues page on GitHub
+- The latest (nightly) wheels
```
Comment (edoakes) [inline doc/source/installation.rst:186]: Maybe we can add some links here?

File: doc/source/installation.rst [hunk 2]
```diff
@@ -67,6 +70,41 @@ For example, here are the Ray 0.9.0.dev0 wheels for Python 3.5, MacOS for commit
...
+.. _windows-support:
+
+Windows Support
+---------------
+
+Windows support is currently limited and "alpha" quality.
+Bugs, process/resource leaks, or other incompatibilities may exist under various scenarios.
+Unusual, unattended, or production usage is **not** recommended.
+
+To use Ray on Windows, Visual C++ runtime must be installed (see :ref:`Windows Dependencies <windows-dependencies>` section).
```
Comment (mehrdadn) [inline doc/source/installation.rst:82]:
```suggestion
To use Ray on Windows, the Visual C++ runtime must be installed (see :ref:`Windows Dependencies <windows-dependencies>` section).
```

File: doc/source/installation.rst (hunk 1 above)
Comment (richardliaw) [reply doc/source/installation.rst]: done

**Pull request merged**